pkg/k8s/api.go

```diff
@@ -109,16 +114,74 @@ func (kubeapi *kubernetesApi) checkApiAccess(client *http.Client) *healthcheckPb
 	return checkResult
 }
 
+// StartProxy opens a proxy connection to Kubernetes, and blocks until the
+// process exits.
+func (kubeapi *kubernetesApi) StartProxy(proxyPort int) error {
+	server, err := proxyCreate(kubeapi.Config)
+	if err != nil {
+		log.Fatalf("proxyCreate() failed with: %+v", err)
```

Comment:
This relates to #17. I don't think we want to call `log.Fatalf` here. That will cause an error message to be logged and then the process will immediately exit with a non-zero exit code. Since the method signature of this method allows you to return an error, I think it would be better to return an error, and let the caller decide how they want to log / exit.

It's also worth noting that the previous implementation of this method used `return fmt.Errorf` instead of `log.Fatalf`.